File: java/p001.java
public class p001 {
    public static void main(String[] args) {
        int lim = 999;
        System.out.println(
            arithmeticSeries(3, lim/3*3, lim/3)
            + arithmeticSeries(5, lim/5*5, lim/5)
            - arithmeticSeries(15, lim/15*15, lim/15)
        );
    }

    public static int arithmeticSeries(int first, int last, int terms) {
        return terms * (first + last) / 2;
    }
}

File: go/p001.go
package main

import "fmt"

func arithmeticSeries(first int, last int, terms int) int {
	return terms * (first + last) / 2
}

func main() {
	limit := 999
	fmt.Println(arithmeticSeries(3, limit/3*3, limit/3) + arithmeticSeries(5, limit/5*5, limit/5) - arithmeticSeries(15, limit/15*15, limit/15))
}
